README.md

```diff
@@ -6,3 +6,188 @@ Distributed service to interact with development boards.
+   ```
+
+1. Copy the [example systemd unit](boardswarm/share/boardswarm.service) to
+   `/etc/systemd/system/boardswarm.service`.
+
+1. Copy [example udev rules](boardswarm/share/99-boardswarm.rules) to
+   `/etc/udev/rules.d/`.
+
+1. Reload udev rules and ensure they are applied:
+   ```
+   $ sudo udevadm control --reload-rules
+   $ sudo udevadm trigger
+   ```
+
+1. Copy example [`server.conf`](boardswarm/share/server.conf) to
+   `/etc/boardswarm/server.conf` on target server machine and modify as
+   required.
+
+### Server Configuration
+
+1. Install [pdudaemon](https://github.com/pdudaemon/pdudaemon) and configure it
+   to switch the development boards on and off, using whichever mechanism you
+   are using (ensure that the `server.conf` `pdu` provider is configured to match
+   the configuration set in the pdudaemon configuration):
+   ```
+   $ sudo apt install pdudaemon
+   ```
+
+1. We are going to use Static JWKS based authentication, as documented in the
+   [server README](boardswarm/README.md#static-jwks-based-authentication).
+   Install `rnbyc` so we can create the required tokens:
+   ```
+   $ sudo apt install rnbyc
+   ```
+
+1. Generate an EdDSA key:
+   ```
+   $ rnbyc -j -g  Ed25519  -x   -o private.jwks -p auth.jwks
+   ```
+   Ensure you keep the private key private and safe, it will be needed during
+   client configuration.
+
+1. Copy generated `auth.jwks` to `/etc/boardswarm/`
+   ```
+   $ sudo cp auth.jwks /etc/boardswarm/
+   ```
+
+For each development board you will need a section under `devices`. There is an
+example one provided in the example server configuration which can be used as a
+template.
+
+#### USB Serial Consoles
+
+1. Determine the USB serial port used as the console on the development board.
+
+1. Use the following commmand to determine the required `udev.ID_SERIAL` and
+   `udev.ID_USB_INTERFACE_NUM`:
+   ```
+   $ udevadm info <tty_device> | grep -i 'ID_SERIAL=\|ID_USB_INTERFACE_NUM='
+   ```
+
+#### DFU Booting
+
+1. Power on the development board in DFU mode and run:
+   ```
+   $ sudo dfu-util -l
+   dfu-util 0.11
+   
+   Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
+   Copyright 2010-2021 Tormod Volden and Stefan Schmidt
+   This program is Free Software and has ABSOLUTELY NO WARRANTY
+   Please report bugs to http://sourceforge.net/p/dfu-util/tickets/
+   
+   Found DFU: [0451:6165] ver=0200, devnum=45, cfg=1, intf=0, path="3-3.4", alt=1, name="SocId", serial="01.00.00.00"
+   Found DFU: [0451:6165] ver=0200, devnum=45, cfg=1, intf=0, path="3-3.4", alt=0, name="bootloader", serial="01.00.00.00"
+   ```
+
+1. Ensure that you have udev rules configuring the group ownership of usb
+   devices with the given USB vendor and product IDs (`0451:6165` in the above
+   example) to the boardswarm group.
+
+1. Use the following command to determine the required `ID_PATH`
+   ```
+   $ udevadm info /sys/bus/usb/devices/<path> | grep 'ID_PATH='
```

Comment:
similar you can get this from `boardswarm-cli list volumes` combined with `boardswarm-cli volume <volume> properties`

Reply:
done